core/engine/src/builtins/regexp/mod.rs

```diff
@@ -943,72 +943,65 @@ impl RegExp {
         // 11. If fullUnicode is true, let input be StringToCodePoints(S). Otherwise, let input be a List whose elements are the code units that are the elements of S.
         // 12. NOTE: Each element of input is considered to be a character.
 
+        // TODO: Comment spec deviation
+        // TODO: It would be better to put this in an enum.
+        //       enum Matches { Utf16(..), Ucs2(..) }
```

Comment:
Couldn't we just do the construction and the call to `next` inline? I'm seeing those are only used in the `r` assignment.

Reply:
Yes, will change that, leftover when there was a loop :sweat_smile: